src/bio.c

```diff
@@ -82,12 +73,24 @@ static unsigned int bio_job_to_worker[] = {
     [BIO_RDB_SAVE] = 3,
 };
 
-static pthread_t bio_threads[BIO_WORKER_NUM];
-static pthread_mutex_t bio_mutex[BIO_WORKER_NUM];
-static pthread_cond_t bio_newjob_cond[BIO_WORKER_NUM];
-static list *bio_jobs[BIO_WORKER_NUM];
+typedef struct {
+    const char *const bio_worker_title;
+    pthread_t bio_thread_id;
+    pthread_mutex_t bio_mutex;
+    pthread_cond_t bio_newjob_cond;
+    list *bio_jobs;
+} bio_worker_data;
+
+static bio_worker_data bio_workers[] = {
+    {"bio_close_file"},
+    {"bio_aof"},
+    {"bio_lazy_free"},
+    {"bio_rdb_save"}};
+static const bio_worker_data *const bio_worker_end = bio_workers + (sizeof bio_workers / sizeof *bio_workers);
+#define BIO_WORKER_NUM(X) ((X) - bio_workers)
```

Comment:
In general I do not find it bad to wrap this as a macro. However I would suggest to make this a static function.
why?
1. macro's expansion is int (IIRC) and in some cases you might want to get an unsigned value out of it and it will force the user to case. a function can return unsigned
2. it might cause compiler warnings if the passes worker is const... probably not something that will happen (unless we pass it the bio_worker_end ) but a function can convert the input pointer correctly IMO.
3. a function is a good place to place asserts and/or return some invalid value in cases the user provides some bad pointer which is not a bio_workers